_posts/2023-05-30-cloud-gov-pages-jekyll-ruby-upgrade.md

```diff
@@ -0,0 +1,52 @@
+---
+layout: post
+title: Upgrading a Jekyll site to Ruby 3.1
+date: May 24, 2023
+excerpt: "Ruby 2.7 has reached “end of life” so we’re providing instructions on how to upgrade your Jekyll site to use Ruby 3.1. Cloud.gov Pages will continue supporting Ruby 2.7 builds for six months, but then only Ruby 3 versions will be supported."
+---
+
+Ruby 2.7 has reached “end of life” so we’re providing instructions on how to upgrade your Jekyll site to use Ruby 3.1. Cloud.gov Pages will continue supporting Ruby 2.7 builds for six months, but then only Ruby 3 versions will be supported. 
+
+The main difficulty with upgrading a site to use a newer version of Ruby is that one or more of your required Gems could break when changing Ruby versions. This is notably the case for `jekyll-assets`. If you don’t have that dependency, this could be a relatively straightforward upgrade. We recently upgraded our own [Jekyll template repo](https://github.com/cloud-gov/pages-uswds-jekyll) which you can check out for reference.
+
+## Instructions for sites without jekyll-assets
+
+If you don’t see `jekyll-assets` in your `Gemfile` or `Gemfile.lock`, you don’t have a dependency on `jekyll-assets` and you might be able to upgrade to Ruby 3.1 with just a few steps:
+- Create a file called `.ruby-version` in your site folder which contains the string `3.1`. If you already have this file, you can replace the previous version with `3.1`
+- If you have any other site-specific scripts which specify the ruby version, update these as well.
+- Delete your previous `Gemfile.lock` and regenerate by running `bundle install`
```

Comment:
Is it necessary to regenerate the lockfile, or is it enough to update the "RUBY VERSION" section? (Aside: Not sure if there's an easy way to do this aside from manually editing the file). Might save the effort of the sub-bullet points as well.

Reply:
My concern was that some other underlying dependency could change in response to the version switch but this also reminds me that the Gemfile needs to have the version specified/changed as well